algebra/group/subgroup.lean

```diff
@@ -0,0 +1,148 @@
+import data.set.basic init.function data.equiv init.logic 
+import algebra.group
+
+open set
+
+universes u v
+variables {G : Type u} {H : Type v}
+
+class subgroup [group G] (S : set G) : Prop := 
+    (mul_mem : ∀ {a b}, a ∈ S → b ∈ S → a * b ∈ S) 
+    (one_mem : (1 : G) ∈ S)
+    (inv_mem : ∀ {a}, a ∈ S → a⁻¹ ∈ S) 
+
+class normal_subgroup [group G] (S : set G) extends subgroup S : Prop :=
+    (normal : ∀ n ∈ S, ∀ g : G, g * n * g⁻¹ ∈ S)
+             
+namespace subgroup
+variable [group G]
+
+attribute [simp] subgroup.one_mem 
+                 subgroup.inv_mem 
+                 subgroup.mul_mem
+                 normal_subgroup.normal
+
+-- Subgroup is a group
+instance (S : set G) [subgroup S] : group S :=
+{   mul := λ ⟨x, hx⟩ ⟨y, hy⟩, ⟨x * y, mul_mem hx hy⟩,
+    mul_assoc := λ ⟨x, hx⟩ ⟨y, hy⟩ ⟨z, hz⟩, subtype.eq $ mul_assoc x y z,
+    one := ⟨1, one_mem S⟩,
+    one_mul := λ ⟨x, hx⟩, subtype.eq $ one_mul x,
+    mul_one := λ ⟨x, hx⟩, subtype.eq $ mul_one x,
+    inv := λ ⟨x, hx⟩, ⟨x⁻¹, inv_mem hx⟩,
+    mul_left_inv := λ ⟨x, hx⟩, subtype.eq $ mul_left_inv x }
```

Comment:
style: 2 indent

Reply:
Fixed in https://github.com/leanprover/mathlib/pull/94/commits/91693ec5a4d0762766254c8a4d3153659349e094.